torch/csrc/autograd/utils/lambda_post_hook.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <torch/csrc/autograd/function_hook.h>
+
+namespace torch {
+namespace autograd {
+namespace utils {
+
+// Turns lambda into a torch::autograd::FunctionPostHook.
+class LambdaPostHook : public torch::autograd::FunctionPostHook {
+  using variable_list = std::vector<torch::autograd::Variable>;
+
+ public:
+  /* implicit */ LambdaPostHook(std::function<void(const variable_list&)> fn)
```

Comment:
While you're changing this signature, why not go all the way and also allow it to modify the output. That way we are future proof and won't have to change this again?